lib/OpenQA/WebAPI/Plugin/REST.pm

```diff
@@ -2,49 +2,33 @@
 # SPDX-License-Identifier: GPL-2.0-or-later
 
 package OpenQA::WebAPI::Plugin::REST;
-use Mojo::Base 'Mojolicious::Plugin';
+use Mojo::Base 'Mojolicious::Plugin', -signatures;
 
-use Scalar::Util ();
-use Carp ();
-
-sub register {
-
-    my ($self, $app, $config) = @_;
+use Scalar::Util 'blessed';
+use Carp 'croak';
 
+sub register ($self, $app, $config) {
     # special anchor tag with data-method
     $app->helper(
-        action_link => sub {
-            my ($self, $method, $content) = (shift, shift, shift);
+        action_link => sub ($self, $method, $content, @args) {
```

Comment:
This `$self` is incorrect, there is no instance to refer to, it's just `$c` again.

Reply:
thx, see https://github.com/os-autoinst/openQA/pull/4419